### PR TITLE
fix a bug when switching listctrl column header image

### PR DESCRIPTION
**Pull request by playgithub**

### CODE REVIEW

File: src/msw/listctrl.cpp [hunk 1]
```diff
@@ -3689,12 +3689,24 @@ static void wxConvertToMSWListCol(HWND hwndList,
...
         // version only draws them on the right (we don't have a flag to
         // specify the image location anyhow)
-        const int fmtImage = LVCFMT_BITMAP_ON_RIGHT | LVCFMT_COL_HAS_IMAGES;
-
-        if ( item.m_image != -1 )
+        const int fmtImage = LVCFMT_IMAGE | LVCFMT_BITMAP_ON_RIGHT; // Not sure if LVCFMT_COL_HAS_IMAGES means the whole column header is an image
+                                                                    // LVCFMT_IMAGE works well
+        if (item.m_image != -1)
+        {
+            lvCol.mask |= LVCF_IMAGE; // Since Windows Common Control Version 4.70,
```
Comment (vadz) [inline src/msw/listctrl.cpp]: I don't think this is needed, it's already set above unconditionally anyhow.

Alternatively, we can keep this one, but remove the line setting it above and the one resetting it below. I think I actually prefer doing it like this...

**Review by vadz — COMMENTED** [review]
Thanks for the fix! I think the really important change is resetting `LVCF_IMAGE`, thanks for finding that this was needed.

As mentioned in the comment below, I'm committing slightly modified form of your changes, but the effect should be the same -- please let me know if you see any problems with it.

Thanks again!

**Comment by vadz** [comment]
Oops, I used wrong PR number in 95eea38eb0d68c1de65a51ae20c61ec29507258f commit message, it should have been this one.

File: src/msw/listctrl.cpp (hunk 1 above)
Comment (playgithub) [reply src/msw/listctrl.cpp]: Sorry for my carelessness.
The way that set LVCF_IMAGE conditionally makes the code looks symmetrical :)

**Pull request closed without merging**